Automattic/jetpack
# API Nonces: Revert of the Runtime Cleanup.

Replaces and closes #17389

#### Changes proposed in this Pull Request:
* Runtime Cleanup functionality will be revisited. Reverting that for the time being.

#### Jetpack product discussion
`p1HpG7-ajF-p2#comment-42552`

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Use the Jetpack Debug's "Mocker" tool to create a bunch of old nonces (a few dozens would be enough).
2. Go to the [Developer Console](https://developer.wordpress.com/docs/api/console/) and send an API request:
`WP.COM API`, `GET`, `v1.2`, `/sites/your-site.com`. It will create a few valid nonces.
3. Count the nonces, write down how many are there:
```sql
SELECT COUNT(*) FROM wp_options WHERE option_name LIKE 'jetpack\_nonce\_%';
```
4. Use the "Crontrol" plugin to run the `jetpack_clean_nonces` hook.
5. Run the `COUNT` SQL query again and confirm that the "mocked" nonces were removed.
6. Fetch the oldest nonce and confirm that it's a fresh one:
```sql
SELECT FROM_UNIXTIME(option_value, '%Y-%m-%d %H:%i:%s') FROM wp_options WHERE option_name LIKE 'jetpack\_nonce\_%' ORDER BY option_value LIMIT 1;
```
7. Disconnect Jetpack.
8. Run the `COUNT` SQL query again and confirm that all the nonces were removed.

#### Proposed changelog entry for your changes:
* Reverting the Nonce Runtime Cleanup functionality.
